geomstats/geometry/euclidean.py

```diff
@@ -86,9 +107,69 @@ def metric_matrix(self, base_point=None):
         inner_prod_mat : array-like, shape=[..., dim, dim]
             Inner-product matrix.
         """
+        if self._space.point_ndim > 1:
+            raise NotImplementedError("`metric_matrix` is not implemented.")
+
+        dim = self._space.dim
+        return repeat_out(
+            self._space.point_ndim,
+            gs.copy(self.metric_matrix_),
+            base_point,
+            out_shape=(dim, dim),
+        )
+
+    def inner_product_derivative_matrix(self, base_point=None):
+        r"""Compute derivative of the inner prod matrix at base point.
+
+        Writing :math:`g_{ij}` the inner-product matrix at base point,
+        this computes :math:`mat_{ijk} = \partial_k g_{ij}`, where the
+        index k of the derivation is put last.
+
+        Parameters
+        ----------
+        base_point : array-like, shape=[..., dim]
+            Base point.
+
+        Returns
+        -------
+        metric_derivative : array-like, shape=[..., dim, dim, dim]
+            Derivative of the inner-product matrix, where the index
+            k of the derivation is last: :math:`mat_{ijk} = \partial_k g_{ij}`.
+        """
+        if self._space.point_ndim > 1:
+            raise NotImplementedError(
+                "`inner_product_derivative_matrix` is not implemented."
+            )
+
+        dim = self._space.dim
+        shape = (dim, dim, dim)
+        return repeat_out(
+            self._space.point_ndim, gs.zeros(shape), base_point, out_shape=shape
+        )
+
+    def christoffels(self, base_point):
+        """Christoffel symbols associated with the connection.
+
+        The contravariant index is on the first dimension.
+
+        Parameters
+        ----------
+        base_point : array-like, shape=[..., dim]
+            Point on the manifold.
+
+        Returns
+        -------
+        gamma : array-like, shape=[..., dim, dim, dim]
+            Christoffel symbols, with the contravariant index on
+            the first dimension.
+        """
+        if self._space.point_ndim > 1:
+            raise NotImplementedError("The Christoffel symbols are not implemented.")
```

Comment:
Are you sure about this? (Meanwhile I'll move this method to the `EuclideanMetric` to avoid the introduction of any conceptual error, as I'll merge #1946 now.)

According to our code, `christoffels` are obtained from a sum of products between `cometric_matrix` and `inner_product_derivative_matrix` (both obtained from `metric_matrix`). In a flat metric, by definition, the `metric_matrix` is independent of the `base_point`, so the `inner_product_derivative_matrix` is null. Therefore, the christoffels should be zero.

The geodesic equation implementation also suggests they should be zero.